src/components/WindowCanvasNavigationControls.js

```diff
@@ -9,6 +10,62 @@ import ns from '../config/css-ns';
  * Represents the viewer controls in the mirador workspace.
  */
 export class WindowCanvasNavigationControls extends Component {
+  /** */
+  constructor(props) {
+    super(props);
+
+    this.state = { canvasNavWidth: null };
+
+    this.canvasNav = React.createRef();
+  }
+
+  /**
+   * Set the necessary widths in state when the component
+   * mounts and bind the state update to window resize
+  */
+  componentDidMount() {
+    this.updateComponentWidth();
+    window.addEventListener('resize', this.updateComponentWidth.bind(this));
+  }
+
+  /**
+   * Update the component width when the visibility changes
+  */
+  componentDidUpdate(prevProps) {
+    const { visible } = this.props;
+
+    if (prevProps.visible !== visible) {
+      this.updateComponentWidth();
+    }
+  }
+
+  /**
+   * Unbind the window resize listener when unmounting the component
+  */
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateComponentWidth.bind(this));
+  }
+
+  /**
+   * Simple utility function to set state (that we can bind on window resize)
+  */
+  updateComponentWidth() {
+    this.setState({
+      canvasNavWidth: (
+        this.canvasNav && this.canvasNav.current && this.canvasNav.current.offsetWidth
+      ),
+    });
+  }
+
+  /**
+   * Determine if canvasNavControls are stacked (based on a hard-coded width)
+  */
+  canvasNavControlsAreStacked() {
+    const { canvasNavWidth } = this.state;
+
+    return (canvasNavWidth && canvasNavWidth <= 253);
```

Comment:
I'm a bit concerned that this may be fragile.  I wanted to check the width of the container against the width of the zoom controls element (to determine if they were stacked or not) but because of how flexbox works the elements widths are not reliable (they only take up the full width when in the column direction).  I ended up needing to track the width of the container, so it didn't make sense to have the addl. code and potential for finicky behavior of also tracking the width of the child controls.  Happy to pair up if folks think we should explore alternate solutions (e.g. abandoning flexbox all-together for these components).

Reply:
Okay, I took a spike at using `AutoSizer` and I couldn't get it to quite work the way that I needed.  Happy to pair up on trying to implement something (to see if my understanding of how to use `AutoSizer` is off).